Sources/StreamVideoSwiftUI/CallingViews/MicrophoneChecker.swift

```diff
@@ -173,3 +173,20 @@ protocol AudioSessionProtocol {
 }
 
 extension AVAudioSession: AudioSessionProtocol {}
+
+/// Provides the default value of the `ThermalStateObserving` protocol.
+public struct MicrophoneCheckerKey: InjectionKey {
+    public static var currentValue: MicrophoneChecker = MicrophoneChecker()
+}
+
+extension InjectedValues {
+
+    public var microphoneChecker: MicrophoneChecker {
```

Comment:
Not sure if this should be public? Any strong arguments for it?

Reply:
I don't have strong arguments! The less public the better!

Reply:
On a second thought, if integrators would like to add more valueLimits, they will need to override the views and provide a different instance. By having this public they will be able to inject a differently configured instance that will be used from SDK's views without further changes. For this reason it's probably better to keep it as public.

Reply:
It actually seems to be already used in the demo app, which prevents building the app: 

❌  /Users/runner/work/stream-video-swift/stream-video-swift/DemoApp/Sources/Components/DemoAppViewFactory.swift:15:15: key path value type 'WritableKeyPath<InjectedValues, MicrophoneChecker>' cannot be converted to contextual type 'KeyPath<InjectedValues, MicrophoneChecker>'

    @Injected(\.microphoneChecker) var microphoneChecker
        ^



❌  /Users/runner/work/stream-video-swift/stream-video-swift/DemoApp/Sources/Views/CustomCallView.swift:12:15: 'microphoneChecker' is inaccessible due to 'internal' protection level

    @Injected(\.microphoneChecker) var microphoneChecker
              ^



❌  /Users/runner/work/stream-video-swift/stream-video-swift/DemoApp/Sources/Views/CustomCallView.swift:12:15: key path value type 'WritableKeyPath<InjectedValues, MicrophoneChecker>' cannot be converted to contextual type 'KeyPath<InjectedValues, MicrophoneChecker>'